src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -150,47 +117,39 @@ extern "C" void QCALLTYPE RuntimeTypeHandle_CreateInstanceForAnotherGenericParam
     END_QCALL;
 }
 
-NOINLINE FC_BOOL_RET IsInstanceOfTypeHelper(OBJECTREF obj, REFLECTCLASSBASEREF refType)
+FCIMPL2(INT32, RuntimeTypeHandle::IsInstanceOfTypeInternal, ReflectClassBaseObject* pTypeUNSAFE, Object *objectUNSAFE)
 {
     FCALL_CONTRACT;
 
-    BOOL canCast = false;
-
-    FC_INNER_PROLOG(RuntimeTypeHandle::IsInstanceOfType);
-
-    HELPER_METHOD_FRAME_BEGIN_RET_ATTRIB_2(Frame::FRAME_ATTR_EXACT_DEPTH|Frame::FRAME_ATTR_CAPTURE_DEPTH_2, obj, refType);
-    canCast = ObjIsInstanceOf(OBJECTREFToObject(obj), refType->GetType());
-    HELPER_METHOD_FRAME_END();
-
-    FC_RETURN_BOOL(canCast);
+    REFLECTCLASSBASEREF refType = (REFLECTCLASSBASEREF)ObjectToOBJECTREF(pTypeUNSAFE);
+    switch (ObjIsInstanceOfCached(objectUNSAFE, refType->GetType()))
```

Comment:
This does not need to be an FCall. We have managed implementation of this cached lookup in https://github.com/dotnet/runtime/blob/main/src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs and https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastCache.cs

Reply:
Some earlier discussion about this: https://github.com/dotnet/runtime/pull/104103#discussion_r1657943673